core/state/statedb.go

```diff
@@ -114,27 +115,39 @@ type SlotChangeList struct {
 	TxIndex             int
 	StateObjectSuicided map[common.Address]struct{}
 	StateChangeSet      map[common.Address]StateKeys
-	BalanceChangeSet    map[common.Address]struct{}
+	BalanceChangeSet    map[common.Address]*big.Int
 	CodeChangeSet       map[common.Address]struct{}
 	AddrStateChangeSet  map[common.Address]struct{}
 	NonceChangeSet      map[common.Address]struct{}
 }
 
+type UnconfirmedReferDetail struct {
+	// StateObjectSuicided map[common.Address]struct{}
+	BalanceChangeReferred map[common.Address]*big.Int
+	CodeChangeReferred    map[common.Address]struct{} // fixme: codehash to compare? it may not necessary
+	// NonceChangeSet      map[common.Address]struct{}  // nonce should be reliable, it is not necessary
+	AddrStateReferred map[common.Address]struct{} // the address of StateObject we refer, or the state change?
```

Comment:
is there any possibility that we add an address to  `AddrStateReferred` and then we get the changed code but not add it to `CodeChangeReferred`.  or do we check it in the conflict check.